typings/index.d.ts

```diff
@@ -226,6 +226,10 @@ export class ApplicationCommand<PermissionsFetchType = {}> extends Base {
   public version: Snowflake;
   public delete(): Promise<ApplicationCommand<PermissionsFetchType>>;
   public edit(data: ApplicationCommandData): Promise<ApplicationCommand<PermissionsFetchType>>;
+  public setName(name: string): Promise<ApplicationCommand<PermissionsFetchType>>;
+  public setDescription(description: string): Promise<ApplicationCommand<PermissionsFetchType>>;
+  public setDefaultPermission(defaultPermission?: boolean): Promise<ApplicationCommand<PermissionsFetchType>>;
+  public setOptions(options: ApplicationCommandOptionData[]): Promise<ApplicationCommand<PermissionsFetchType>>;
```

Comment:
```suggestion
  public setName(name: string): Promise<this>;
  public setDescription(description: string): Promise<this>;
  public setDefaultPermission(defaultPermission?: boolean): Promise<this>;
  public setOptions(options: ApplicationCommandOptionData[]): Promise<this>;
```

Reply:
It should return the same thing as the edit method

Reply:
This isn't a builder it's a manager.

Reply:
What exactly is the difference though?

Reply:
> What exactly is the difference though?

Managers are backed by the api, whereas builders only output json that _may_ be used with the api later on

Reply:
No, like what is the difference between `this` and `ApplicationCommand` in this case?

Reply:
`this` doesn't take generic parameters whilst `ApplicationCommand` does.

Reply:
But `this` is already an `ApplicationCommand<PermissionsFetchType>` right?

Reply:
> But `this` is already an `ApplicationCommand<PermissionsFetchType>` right?

If so, shouldn't it be `typeof this`? Because `this` is an object, not a type.

Reply:
`this` is always a type, it just represents the type of the enclosing class.